functional-samples/sample.sidepanel-dictionary/sidepanel.js

```diff
@@ -19,14 +19,27 @@ const words = {
     "A UI surface which appears when an extension's action icon is clicked."
 };
 
-chrome.runtime.onMessage.addListener(({ name, data }) => {
-  if (name === 'define-word') {
-    // Hide instructions.
-    document.body.querySelector('#select-a-word').style.display = 'none';
+chrome.storage.session.get('lastWord', ({ lastWord }) => {
+  updateDefinition(lastWord);
+});
 
-    // Show word and definition.
-    document.body.querySelector('#definition-word').innerText = data.value;
-    document.body.querySelector('#definition-text').innerText =
-      words[data.value.toLowerCase()];
+chrome.storage.session.onChanged.addListener((changes) => {
+  if ('lastWord' in changes) {
+    updateDefinition(changes['lastWord'].newValue);
```

Comment:
optional: I prefer avoiding using the string literal twice

```
const lastWord = changes['lastWord'];
if (!lastWord) {
  return
}
updateDefinition(lastWord.newValue);
```

Reply:
Gave a change a try, let me know if that looks better.

Reply:
I think it looks better :-) - thanks!